patches/acorn+7.1.1.patch

```diff
@@ -0,0 +1,17 @@
+diff --git a/node_modules/acorn/dist/acorn.js b/node_modules/acorn/dist/acorn.js
+index e2b3317..e8cc8f9 100644
+--- a/node_modules/acorn/dist/acorn.js
++++ b/node_modules/acorn/dist/acorn.js
+@@ -1811,7 +1811,11 @@
+       if (checkClashes) {
+         if (has(checkClashes, expr.name))
+           { this.raiseRecoverable(expr.start, "Argument name clash"); }
+-        checkClashes[expr.name] = true;
++          if (expr.name === 'constructor') {
++            console.log('@@@cannot assign to .constructor property in checkClashes');
++          } else {
++            checkClashes[expr.name] = true;
++          }
```

Comment:
Good find!  Something like this should work:

```suggestion
-        checkClashes[expr.name] = true;
+       Object.defineProperty(checkClashes, expr.name, { value: true });
```

Reply:
Not clear from the suggested change (I got the call wrong too), but you can replace the original:

```js
          checkClashes[expr.name] = true;
```

with

```js
       Object.defineProperty(checkClashes, expr.name, { configurable: true, enumerable: true, value: true });
```